tests/api_workflow/test_api_workflow_datasources.py

```diff
@@ -17,351 +18,591 @@
-    with pytest.raises(ValueError):
-        client.download_raw_metadata(run_id="foo")
-    # should raise ValueError when only relevant_filenames_artifact_id is given
-    with pytest.raises(ValueError):
-        client.download_raw_metadata(relevant_filenames_artifact_id="bar")
-
-
-def test_get_processed_until_timestamp(mocker: MockerFixture) -> None:
-    mocker.patch.object(ApiWorkflowClient, "__init__", return_value=None)
-    mocked_datasources_api = mocker.MagicMock()
-    client = ApiWorkflowClient()
-    client._dataset_id = "dataset-id"
-    client._datasources_api = mocked_datasources_api
-    client.get_processed_until_timestamp()
-    mocked_method = (
-        mocked_datasources_api.get_datasource_processed_until_timestamp_by_dataset_id
-    )
-    mocked_method.assert_called_once_with(dataset_id="dataset-id")
-
-
-def test_set_azure_config(mocker: MockerFixture) -> None:
-    mocker.patch.object(ApiWorkflowClient, "__init__", return_value=None)
-    mocked_datasources_api = mocker.MagicMock()
-    client = ApiWorkflowClient()
-    client._datasources_api = mocked_datasources_api
-    client._dataset_id = "dataset-id"
-    client.set_azure_config(
-        container_name="my-container/name",
-        account_name="my-account-name",
-        sas_token="my-sas-token",
-        thumbnail_suffix=".lightly/thumbnails/[filename]-thumb-[extension]",
-    )
-    kwargs = mocked_datasources_api.update_datasource_by_dataset_id.call_args[1]
-    assert isinstance(
-        kwargs["datasource_config"].actual_instance, DatasourceConfigAzure
-    )
+        # should raise ValueError when only run_id is given
+        with pytest.raises(ValueError):
+            next(client.download_raw_metadata_iter(run_id="run-id"))
+
+        # should raise ValueError when only relevant_filenames_artifact_id is given
+        with pytest.raises(ValueError):
+            next(
+                client.download_raw_metadata_iter(
+                    relevant_filenames_artifact_id="relevant-filenames",
+                )
+            )
+
+    def test_download_new_raw_samples(self, mocker: MockerFixture) -> None:
+        client = ApiWorkflowClient(token="abc", dataset_id="dataset-id")
+        client.get_processed_until_timestamp = mocker.MagicMock(return_value=2)
+        mocker.patch("time.time", return_value=5)
+        client.download_raw_samples = mocker.MagicMock()
+        client.update_processed_until_timestamp = mocker.MagicMock()
+        client.download_new_raw_samples()
+        client.download_raw_samples.assert_called_once_with(
+            from_=2 + 1,
+            to=5,
+            relevant_filenames_file_name=None,
+            use_redirected_read_url=False,
+        )
+        client.update_processed_until_timestamp.assert_called_once_with(timestamp=5)
+
+    def test_download_new_raw_samples__from_beginning(
+        self, mocker: MockerFixture
+    ) -> None:
+        client = ApiWorkflowClient(token="abc", dataset_id="dataset-id")
```

Comment:
The old code was patching `__init__`. Are you sure it is not necessary? It seems to me that init does an api call (version check) which we don't want to do in tests.

Reply:
Yes, this is not necessary because the versioning api is disable in all tests: https://github.com/lightly-ai/lightly/blob/a0f4bb047ac9d19340d7d794425e8e9d14a22e03/tests/conftest.py#L45